### ISSUE TITLE
False positive unsafe function call warning for vec![] on nightly

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust-analyzer#18871]
**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

**rustc version**: `rustc 1.86.0-nightly (1891c2866 2025-01-04)`

**editor or extension**: VSCode

**relevant settings**: none

**repository link (if public, optional)**:

**code snippet to reproduce**: 
```rust
fn foo() {
   vec![0]; //~ call to unsafe function is unsafe and requires an unsafe function or block
}
```

This happens because `vec![]` now expands to a `alloc::boxed::box_new` call which is marked as `#[rustc_intrinsic]`. Rust-analyzer however doesn't treat it as safe intrinsic. In the past there was a list of hard coded intrinsics that are safe, but nowadays any `#[rustc_intrinsic]` which doesn't have `unsafe` as part of their signature is safe (rustc still double checks the declaration I think, but rust-analyzer shouldn't need to).
